src/Microsoft.TemplateEngine.Cli/New3Command.cs

```diff
@@ -529,16 +537,26 @@ private CreationResultStatus EnterMaintenanceFlow()
                 argsError = !ValidateRemainingParameters();
             }
 
-            if (argsError || IsHelpFlagSpecified)
+            if(IsHelpFlagSpecified)
             {
-                if(commandParseFailureMessage != null)
+                if (commandParseFailureMessage != null)
                 {
                     Reporter.Error.WriteLine(commandParseFailureMessage.Bold().Red());
                 }
 
                 ShowUsageHelp();
                 ShowTemplateHelp(UnambiguousTemplateToUse.Info);
-                return argsError ? CreationResultStatus.InvalidParamValues : CreationResultStatus.Success;
+                return CreationResultStatus.Success;
```

Comment:
I think having it be an error helps for scripting cases even though I'm not sure why someone would specify --help in that case.